### DISCUSSION TITLE
I liked the project, congratulations!

### DISCUSSION THREAD

**Discussion by BlackMix** [hotspotbilling/phpnuxbill#73] [Ideas]
I liked the project, congratulations.
I'm studying the project and I've seen that it can be improved in:

The customer's account does not need to add an address, the address is in the services contract for the customer, in which case they may have several services with speed plans for different addresses, in the services they can add geolocation which can enable them to find the address faster if they need support.

example:
```
tbl_customers
tbl_customers_services
tbl_customers_address
```

In this case, there would be small changes but it would be very good for better control of payments and well organized to legalize the project in the future, so it would go head to head with paid projects.

I hope my ideas are welcome, I really liked the project.

**Comment by ibnux** [comment]
I already add **tbl_customers_meta**

It can be used for additional attributes
Just like WordPres

But not yet implemented

https://github.com/hotspotbilling/phpnuxbill/blob/master/install/phpnuxbill.sql#L36-L41

**Reply by BlackMix** [reply]
Nice.

I'm studying the project, I've never seen this way of programming like that project, I'm accustomed to Laravel, Vuejs and Reactjs haha


This Meta table is interesting but for future filter searches it will be a bit complicated to use, the interesting thing is having the table for information for services/contracts and separate addresses, it even improves performance..

Recently I saw a project for a provider with 100 thousand users and it takes too long to load customer information because of low performance implementations, and it's a paid system lol, the open source system is better than these paid systems I've seen

**Reply by BlackMix** [reply]
I'm building an ISP from scratch, I bought Mikrotik equipments, I'm going to use this system and test, need make one gateway for Santander bank, but easy to do it.
